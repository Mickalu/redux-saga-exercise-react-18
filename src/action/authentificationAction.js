export const authentificationAction = {
  REGISTER_USER: "REGISTER_USER",
  UPDATE_STATUS_REGISTER_USER: "UPDATE_STATUS_REGISTER_USER",
  UPDATE_DATA_REGISTER_USER: "UPDATE_DATA_REGISTER_USER",
};
