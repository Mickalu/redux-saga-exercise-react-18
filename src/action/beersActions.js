export const beersActions = {
  FETCH_BEERS: "FETCH_BEERS",
};
