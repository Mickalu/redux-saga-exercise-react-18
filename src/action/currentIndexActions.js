export const currentIndexActions = {
  INCREMENT_CURRENT_INDEX: "INCREMENT_CURRENT_INDEX",
  ADD_BEER_LIKED_BEERS: "ADD_BEER_LIKED_BEERS",
};
