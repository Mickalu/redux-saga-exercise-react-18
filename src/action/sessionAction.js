export const sessionActions = {
  SESSION_START: 'SESSION_START',
};
