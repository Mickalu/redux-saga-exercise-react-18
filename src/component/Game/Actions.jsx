import React from 'react';

import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import Container from 'react-bootstrap/Container';
import { faHeart, faXmark } from '@fortawesome/free-solid-svg-icons';

const Actions = ({ passNextBeer, likeBeer }) => {

  return (
    <Container data-testid="actions-container" style={{ background: 'white', borderRadius: '0 0 25px 25px' }}>
      <ul  className="list-unstyled list-inline text-center game-actions">
        <li>
          <button
            type="button"
            className="btn btn-danger btn-circle btn-xl"
            name="dislikeBtn"
            onClick={() => passNextBeer()}
            data-testid="dislike-button"
          >
            <FontAwesomeIcon icon={faXmark} />
          </button>
        </li>
        <li />
        <li>
          <button
            type="button"
            className="btn btn-success btn-circle btn-xl"
            name="likeBtn"
            onClick={() => likeBeer()}
            data-testid="like-button"
          >
            <FontAwesomeIcon icon={faHeart} />
          </button>
        </li>
      </ul>
    </Container>
  );
};

export default Actions;
