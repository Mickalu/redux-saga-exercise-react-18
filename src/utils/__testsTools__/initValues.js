export const initListBeers = {
  data:
    [
      {
        id: "82",
        title: "Karmeliet Triple",
        description: "<p>Do we really have to introduce Triple Karmeliet, our best-selling beer?</p>",
        country: "Belgium",
        type_beer: "Tripel",
        company: "Brasserie Bosteels",
        colour: "Blond",
        graduation: "6 %",
        size: "33 cl",
        price: "2.5",
        photo_link: "https://img.saveur-biere.com/img/p/82-14662-thickbox_sb3.jpg",
      },
      {
        id: "126",
        title: "Cuvée des trolls",
        description: "<p>A tradionally made beer with fruity notes of orange and honey. Discover one of the reference beers from Belgium!</p>",
        country: "Belgium",
        type_beer: "Belgian Pale ale / Blonde",
        company: "Brasserie Dubuisson",
        colour: "Blond",
        graduation: "8 %",
        size: "25 cl",
        price: "1.6",
        photo_link: "https://img.saveur-biere.com/img/p/126-14990-thickbox_sb3.jpg",
      },
      {
        id: "104",
        title: "Kwak",
        description: "<p>This Belgian beer no longer needs any introduction. Best poured in the original coachman's glass from this brand!</p>",
        country: "Belgium",
        type_beer: "Belgian Pale ale / Blonde",
        company: "Brasserie Bosteels",
        colour: "Amber",
        graduation: "8 %",
        size: "33 cl",
        price: "2.1",
        photo_link: "https://img.saveur-biere.com/img/p/104-16465-thickbox_sb3.jpg",
      },
      {
        id: "81",
        title: "Orval",
        description: "<p>Orval is a triple trappist bitter-sweet beer with an evolutive flavour! A true Belgian reference beer.</p>",
        country: "Belgium",
        type_beer: "Belgian Pale ale / Blonde",
        company: "Abbaye d'Orval",
        colour: "Amber",
        graduation: "6 %",
        size: "33 cl",
        price: "2.8",
        photo_link: "https://img.saveur-biere.com/img/p/81-14420-thickbox_sb3.jpg",
      },
      {
        id: "91",
        title: "Rochefort 10",
        description: "<p>A Belgian Trappist beer with a plum and cacao taste! A Marvellous product with a 11.3° ABV. </p>",
        country: "Belgium",
        type_beer: "Quadrupel",
        company: "Abbaye de Rochefort",
        colour: "Dark",
        graduation: "10 %",
        size: "33 cl",
        price: "3.4",
        photo_link: "https://img.saveur-biere.com/img/p/91-14652-thickbox_sb3.jpg",
      },
      {
        id: "101",
        title: "Lindemans Faro",
        description: "<p>Discover a Belgian fruit beer, extremely refreshing and sweet. Maybe the most popular beer among women!</p>",
        country: "Belgium",
        type_beer: "Faro",
        company: "Brasserie Lindemans",
        colour: "Amber",
        graduation: "8 %",
        size: "25 cl",
        price: "1.8",
        photo_link: "https://img.saveur-biere.com/img/p/101-14699-thickbox_sb3.jpg",
      },
      {
        id: "100",
        title: "Lindemans Pêcheresse",
        description: "<p>Discover this beer with peach flavors to be poured in a flute-shaped glass. A nice cocktail for the ladies!</p>",
        country: "Belgium",
        type_beer: "Lambic - Fruit",
        company: "Brasserie Lindemans",
        colour: "Blond",
        graduation: "9 %",
        size: "25 cl",
        price: "2.2",
        photo_link: "https://img.saveur-biere.com/img/p/100-14012-thickbox_sb3.jpg",
      },
      {
        id: "736",
        title: "Lindemans Kriek",
        description: "<p>Discover a beer flavored with cherries, highly popular among women!</p>",
        country: "Belgium",
        type_beer: "Lambic - Fruit",
        company: "Brasserie Lindemans",
        colour: "Ruby",
        graduation: "4 %",
        size: "25 cl",
        price: "1.9",
        photo_link: "https://img.saveur-biere.com/img/p/736-14700-thickbox_sb3.jpg",
      },
      {
        id: "99",
        title: "Lindemans Framboise",
        description: "<p>Discover a beer with a strong taste of raspberry and a fine balance between acidity and fruity sweetness!</p>",
        country: "Belgium",
        type_beer: "Lambic - Fruit",
        company: "Brasserie Lindemans",
        colour: "Ruby",
        graduation: "6 %",
        size: "25 cl",
        price: "2.1",
        photo_link: "https://img.saveur-biere.com/img/p/99-14010-thickbox_sb3.jpg",
      },
      {
        id: "58",
        title: "Kasteel Bruin",
        description: "<p>Discover a strong Belgian beer with roasted notes, hints of moka, liquorice and banana!</p>\n<p> </p>",
        country: "Belgium",
        type_beer: "Belgian dark ale / Brune",
        company: "Brasserie Van Honsenbrouck",
        colour: "Dark",
        graduation: "9 %",
        size: "33 cl",
        price: "2.6",
        photo_link: "https://img.saveur-biere.com/img/p/58-16778-thickbox_sb3.jpg",
      },
      {
        id: "90",
        title: "Rochefort 8",
        description: "<p>An exceptional Belgian Trappist beer with, jammy stone-fruit and spice flavours</p>",
        country: "Belgium",
        type_beer: "Belgian dark ale / Brune",
        company: "Abbaye de Rochefort",
        colour: "Dark",
        graduation: "9 %",
        size: "33 cl",
        price: "2.7",
        photo_link: "https://img.saveur-biere.com/img/p/90-16527-thickbox_sb3.jpg",
      },
      {
        id: "103",
        title: "Honey Barbar",
        description: "<p>Taste all the flavours and the charm of sweet honey in this Honey Barbar.</p>",
        country: "Belgium",
        type_beer: "Belgian Pale ale / Blonde",
        company: "Brasserie Lefebvre",
        colour: "Blond",
        graduation: "7 %",
        size: "33 cl",
        price: "2.4",
        photo_link: "https://img.saveur-biere.com/img/p/103-16445-thickbox_sb3.jpg",
      },
      {
        id: "110",
        title: "Delirium Tremens",
        description: "<p>The most successful product of the brewery Huyghe in a 33cl bottle !</p>",
        country: "Belgium",
        type_beer: "Belgian Pale ale / Blonde",
        company: "Brasserie Huyghe",
        colour: "Blond",
        graduation: "6 %",
        size: "33 cl",
        price: "3.2",
        photo_link: "https://img.saveur-biere.com/img/p/110-16755-thickbox_sb3.jpg",
      },
    ],
};

export const initBeersState = {
  beers: {
    data: [
      {id:"82", title:"Karmeliet Triple", description:"<p>Do we really have to introduce Triple Karmeliet, our best-selling beer?</p>", country:"Belgium", type_beer:"Tripel", company:"Brasserie Bosteels", colour:"Blond", graduation:"6 %", size:"33 cl", price:"2.5", photo_link:"https://img.saveur-biere.com/img/p/82-14662-thickbox_sb3.jpg"},
      {id:"126", title:"Cuvée des trolls", description:"<p>A tradionally made beer with fruity notes of orange and honey. Discover one of the reference beers from Belgium!</p>", country:"Belgium", type_beer:"Belgian Pale ale / Blonde", company:"Brasserie Dubuisson", colour:"Blond", graduation:"8 %", size:"25 cl", price:"1.6", photo_link:"https://img.saveur-biere.com/img/p/126-14990-thickbox_sb3.jpg"}
    ],
    isFetching: false,
  },
};

export const initCurrentIndexState = {
  currentIndex: 0,
};

export const initCurrentIndexLastindex = (listTest) => (
  { currentIndex: listTest.length - 1}
);

export const initBeerState = {
  id:"82", title:"Karmeliet Triple", description:"<p>Do we really have to introduce Triple Karmeliet, our best-selling beer?</p>", country:"Belgium", type_beer:"Tripel", company:"Brasserie Bosteels", colour:"Blond", graduation:"6 %", size:"33 cl", price:"2.5", photo_link:"https://img.saveur-biere.com/img/p/82-14662-thickbox_sb3.jpg"
};

export const initBeersSliceState = {
  data: [],
  isFetching: false,
};
