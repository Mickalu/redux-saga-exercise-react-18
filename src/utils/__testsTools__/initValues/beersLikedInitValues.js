export const INIT_BEERS_LIKED_STATE_VALUE = {
  data: [],
};

export const beersLikedStateWithIds = (listIds) => ({data: listIds});
