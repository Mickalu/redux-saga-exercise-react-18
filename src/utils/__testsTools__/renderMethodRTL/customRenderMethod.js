import React from "react";
import { render, screen } from "@testing-library/react";
import { Provider } from "react-redux";

import { store } from "../../../store";

const getByTextCustom = (text) => (
  screen.getByText(text)
);

const queryAllByRoleCustom = (role, options) => (
  screen.queryAllByRole(role, options)
);

const queryByTestIdCustom = (testId) => (
  screen.queryByTestId(testId)
);

const getByTestIdCustom = (testId) => (
  screen.getByTestId(testId)
);

const getByDisplayValueCustom = (displayValue) => (
  screen.getByDisplayValue(displayValue)
);

const renderWithProviders = ({ children }) => (
  <Provider store={store}>
    { children }
  </Provider>
);

export const renderCustom = (ui, options = {}) => render(ui, { wrapper: renderWithProviders, ...options });

// re-export everything
export * from '@testing-library/react';

export { renderCustom as render };
export { getByTextCustom as getByText };
export { queryAllByRoleCustom as queryAllByRole };
export { queryByTestIdCustom as queryBytestId };
export { getByTestIdCustom as getByTestId };
export { getByDisplayValueCustom as getByDisplayValue };
