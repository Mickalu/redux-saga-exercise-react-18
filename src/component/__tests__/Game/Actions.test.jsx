import React from "react";
import { useDispatch, useSelector } from 'react-redux';

import Actions from "../../Game/Actions";
import { render, getByTestId, fireEvent } from "../../../utils/__testsTools__/renderMethodRTL/customRenderMethod";
import { currentIndexActions } from "../../../action/currentIndexActions";
import { initListBeers } from "../../../utils/__testsTools__/initValues";

const reactRedux = { useDispatch, useSelector };

jest.mock("react-redux", () => ({
  ...jest.requireActual("react-redux"),
  useDispatch: jest.fn(),
  useSelector: jest.fn(),
}));

const likeBeerFunction = jest.fn();
const dislikeBeerFunction = jest.fn();

const dispatchMock = (value) => {
  switch (value.type){
    case currentIndexActions.ADD_BEER_LIKED_BEERS:
      likeBeerFunction();
      break;

    case currentIndexActions.REMOVE_BEER_LIKED_BEERS:
      dislikeBeerFunction();
      break;

    default:
      jest.fn();
  };
};

const useSelectorMock = jest.spyOn(reactRedux, "useSelector");
const useDispatchMock = jest.spyOn(reactRedux, "useDispatch");

useSelectorMock.mockImplementation(callback => {
  return callback({
    beers: initListBeers,
    currentIndex: { currentIndex: 0 },
    beersLiked: [],
  });
});

useDispatchMock.mockReturnValue(jest.fn(value => dispatchMock(value)));

it("Should have a button like", () => {
  render(<Actions />);

  expect(getByTestId("like-button")).toBeInTheDocument();
});

it("Should have a dislike button", () => {
  render(<Actions />);

  expect(getByTestId("dislike-button")).toBeInTheDocument();
});

it("Should activate liked beers if click on like button", () => {
  render(<Actions />);
  fireEvent.click(getByTestId("like-button"));

  expect(likeBeerFunction).toHaveBeenCalledTimes(1);
});

it("Should activate disliked beers if click on dislike button", () => {
  render(<Actions />);
  fireEvent.click(getByTestId("dislike-button"));

  expect(dislikeBeerFunction).toHaveBeenCalledTimes(1);
});

it("Should match with snapShot", () => {
  const useDispatchMock = jest.spyOn(reactRedux, 'useDispatch');

  useDispatchMock.mockReturnValue(jest.fn());

  const wrapper = render(<Actions />);

  expect(wrapper).toMatchSnapshot();
});
