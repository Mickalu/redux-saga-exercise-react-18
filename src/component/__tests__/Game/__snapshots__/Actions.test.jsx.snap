// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match with snapShot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
        data-testid="actions-container"
        style="background: white; border-radius: 0 0 25px 25px;"
      >
        <ul
          class="list-unstyled list-inline text-center game-actions"
        >
          <li>
            <button
              class="btn btn-danger btn-circle btn-xl"
              data-testid="dislike-button"
              name="dislikeBtn"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-xmark "
                data-icon="xmark"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 320 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M310.6 150.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L160 210.7 54.6 105.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L114.7 256 9.4 361.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L160 301.3 265.4 406.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L205.3 256 310.6 150.6z"
                  fill="currentColor"
                />
              </svg>
            </button>
          </li>
          <li />
          <li>
            <button
              class="btn btn-success btn-circle btn-xl"
              data-testid="like-button"
              name="likeBtn"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-heart "
                data-icon="heart"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M47.6 300.4L228.3 469.1c7.5 7 17.4 10.9 27.7 10.9s20.2-3.9 27.7-10.9L464.4 300.4c30.4-28.3 47.6-68 47.6-109.5v-5.8c0-69.9-50.5-129.5-119.4-141C347 36.5 300.6 51.4 268 84L256 96 244 84c-32.6-32.6-79-47.5-124.6-39.9C50.5 55.6 0 115.2 0 185.1v5.8c0 41.5 17.2 81.2 47.6 109.5z"
                  fill="currentColor"
                />
              </svg>
            </button>
          </li>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
      data-testid="actions-container"
      style="background: white; border-radius: 0 0 25px 25px;"
    >
      <ul
        class="list-unstyled list-inline text-center game-actions"
      >
        <li>
          <button
            class="btn btn-danger btn-circle btn-xl"
            data-testid="dislike-button"
            name="dislikeBtn"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-xmark "
              data-icon="xmark"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 320 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M310.6 150.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L160 210.7 54.6 105.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L114.7 256 9.4 361.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L160 301.3 265.4 406.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L205.3 256 310.6 150.6z"
                fill="currentColor"
              />
            </svg>
          </button>
        </li>
        <li />
        <li>
          <button
            class="btn btn-success btn-circle btn-xl"
            data-testid="like-button"
            name="likeBtn"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-heart "
              data-icon="heart"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M47.6 300.4L228.3 469.1c7.5 7 17.4 10.9 27.7 10.9s20.2-3.9 27.7-10.9L464.4 300.4c30.4-28.3 47.6-68 47.6-109.5v-5.8c0-69.9-50.5-129.5-119.4-141C347 36.5 300.6 51.4 268 84L256 96 244 84c-32.6-32.6-79-47.5-124.6-39.9C50.5 55.6 0 115.2 0 185.1v5.8c0 41.5 17.2 81.2 47.6 109.5z"
                fill="currentColor"
              />
            </svg>
          </button>
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
